### PR TITLE
Add RxAutoCompleteTextView.itemClickEvents

### PR DESCRIPTION
**Pull request by vanniktech**
About the Unit tests I could not find a way to properly test it since performCompletion is private in AutoCompleteTextView. Do you have any ideas for that?

### CODE REVIEW

**Comment by vanniktech** [comment]
@JakeWharton should I squash the two commits into one or is this for now good enough?

**Comment by JakeWharton** [comment]
Please squash

File: rxbinding/src/main/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextView.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+     */
+    @CheckResult
+    @NonNull
+    public static Observable<AdapterViewItemClickEvent> itemClickEvents(@NonNull AutoCompleteTextView view) {
+        return Observable.create(new AutoCompleteTextViewItemClickEventOnSubscribe(view));
+    }
+
+    private RxAutoCompleteTextView() {
+        throw new AssertionError("No instances.");
+    }
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextView.java]: This file should be indented using 2 spaces, not 4.

File: rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTestActivity.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public final class RxAutoCompleteTextViewTestActivity extends Activity {
+  AutoCompleteTextView autoCompleteTextView;
+
+  List<String> values;
+  ArrayAdapter<String> adapter;
+
+  @Override protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    values = new ArrayList<>(Arrays.asList("One", "Two", "Three"));
```
Comment (JakeWharton) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTestActivity.java]: You don't need `new ArrayList`. Just setting the value to `Arrays.asList` is enough

File: rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTestActivity.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import android.widget.AutoCompleteTextView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class RxAutoCompleteTextViewTestActivity extends Activity {
+  AutoCompleteTextView autoCompleteTextView;
+
+  List<String> values;
```
Comment (JakeWharton) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTestActivity.java]: This also doesn't need to be a field. Could be a local variable in the `onCreate` method.

File: rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  @Test public void itemClickEvents() {
+    RecordingObserver<AdapterViewItemClickEvent> o = new RecordingObserver<>();
+    Subscription subscription = RxAutoCompleteTextView.itemClickEvents(autoCompleteTextView) //
+      .subscribeOn(AndroidSchedulers.mainThread()) //
+      .subscribe(o);
+      o.assertNoMoreEvents();
+
+    // Don't really know how to test it properly since performCompletion is private in AutoCompleteTextView
```
Comment (JakeWharton) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxAutoCompleteTextViewTest.java]: We can use Espresso to type and click on suggestions. I can work on this part after you fix up the other comments.

**Comment by vanniktech** [comment]
@JakeWharton applied your suggestions and squashed it into one

**Comment by JakeWharton** [comment]
Fixed the test. See https://github.com/JakeWharton/RxBinding/pull/152

**Pull request closed without merging**